timeout/timeout.c

```diff
@@ -510,14 +509,12 @@ int parse_interval(const char *ss, struct TClock *interval) {
 		}
 	}
 
-	i = afterchar(s, '.');
-	if (afterpoint == NULL) {
+	if ((afterpoint = strchr(s, '.')) != NULL) {
+		i = (afterpoint - s);
 		if (i != 0 && s[0] != '.') {
-			afterpoint = &s[i + 1];
 			s[i] = '\0';
```

Comment:
Instead you could do the following:
```
if (afterpoint != s) {
    *afterpoint = '\0';
    afterpoint++;
}
```
and not use `i`. Also, `s[0] != '.'` is probably redundant.

Reply:
There's also this:

>        The strchr() function returns a pointer to the first occurrence of the character c in the string s.

Yeah, I will finish this later today.
Merry Christmas again, by the way.

Reply:
This would go inside the `if ((afterpoint = strchr(s, '.')) != NULL)` I think